apps/schema/src/components/SchemaApp.vue

```diff
@@ -0,0 +1,243 @@
+            this.showLogin = true;
+          } else {
+            this.graphqlError = error.response.errors[0].message;
+          }
+          this.warning = null;
+          this.loading = false;
+        });
+      this.loading = false;
+    },
+    loadSchema() {
+      console.log("load schema");
+      this.graphqlError = null;
+      this.loading = true;
+      this.dirty = false;
+      request(
+        "graphql",
+        "{_session{schemas,roles}_schema{name,tables{name,tableType,inherit,externalSchema,description,semantics,columns{name,table,position,columnType,inherited,key,refSchema,refTable,refLink,refBack,required,description,semantics,validation,visible}}}}"
+      )
+        .then((data) => {
+          let _schema = this.addOldNamesAndRemoveMeta(data._schema);
+          this.schema = this.convertToSubclassTables(_schema);
+          this.schemaNames = data._session.schemas;
+        })
+        .catch((error) => {
+          this.graphqlError = error;
+          if (error.response.status === 400) {
+            this.graphqlError = "Schema not found. Do you need to login?";
+          }
+        });
+      this.loading = false;
+    },
+    addOldNamesAndRemoveMeta(schema) {
+      if (schema) {
+        if (schema.tables) {
+          let tables = schema.tables.filter(
+            (table) => table.tableType !== "ONTOLOGIES"
+          );
+          tables.forEach((t) => {
+            t.oldName = t.name;
+            if (t.columns) {
+              t.columns = t.columns
+                .filter((c) => !c.name.startsWith("mg_"))
+                .map((c) => {
+                  c.oldName = c.name;
+                  return c;
+                })
+                .filter((c) => !c.inherited);
+            } else {
+              t.columns = [];
+            }
+          });
+          schema.tables = tables;
+        } else {
+          schema.tables = [];
+        }
+      }
+      return schema;
+    },
+    convertToSubclassTables(schema) {
+      //columns of subclasses should be put in root tables, sorted by position
+      // this because position can only edited in context of root table
+      schema.tables.forEach((table) => {
+        if (table.inherit === undefined) {
+          this.subclassTables(schema, table.name).forEach((subclass) => {
+            //get columns from subclass tables
+            table.columns.push(...subclass.columns);
+            subclass.columns = [];
+            //add subclass to root table
+            if (!table.subclasses) {
+              table.subclasses = [subclass];
+            } else {
+              table.subclasses.push(subclass);
+            }
+          });
+        }
+        //sort
+        table.columns.sort((a, b) => a.position - b.position);
+      });
+      //remove the subclass tables
+      schema.tables = schema.tables.filter(
+        (table) => table.inherit === undefined
+      );
+      return schema;
```

Comment:
don't like the input modification combined with the returning of it at the end. Since the return value is assigned, it'd be best to not modify the input. This would also make it easier to test if we want unit tests.